solidity-v1/dashboard/src/components/modal/threshold/WithdrawGrantedTokens.jsx

```diff
@@ -0,0 +1,221 @@
+import React, { useMemo, useState } from "react"
+import { withBaseModal } from "../withBaseModal"
+import { ModalBody, ModalFooter, ModalHeader } from "../Modal"
+import Button, { SubmitButton } from "../../Button"
+import useReleaseTokens from "../../../hooks/useReleaseTokens"
+import {
+  Accordion,
+  AccordionItem,
+  AccordionItemButton,
+  AccordionItemHeading,
+  AccordionItemPanel,
+  AccordionItemState,
+} from "react-accessible-accordion"
+import { KEEP } from "../../../utils/token.utils"
+import TokenAmount from "../../TokenAmount"
+import { shortenAddress } from "../../../utils/general.utils"
+import moment from "moment"
+import { add, eq } from "../../../utils/arithmetics.utils"
+import OnlyIf from "../../OnlyIf"
+import List from "../../List"
+import * as Icons from "../../Icons"
+import { useSelector } from "react-redux"
+
+export const WithdrawGrantedTokens = withBaseModal(({ onClose }) => {
+  const DEFAULT_GRANTS_TO_DISPLAY = 5
+
+  const { grants: allGrants } = useSelector((state) => state.tokenGrants)
+
+  const grants = useMemo(() => {
+    return allGrants.filter((grant) => !eq(grant.amount, grant.released))
+  }, [allGrants])
+
+  const [numberOfGrantsDisplayed, setNumberOfGrantsDisplayed] = useState(
+    DEFAULT_GRANTS_TO_DISPLAY
+  )
+  const [selectedGrant, setSelectedGrant] = useState(
+    grants.length === 1 ? grants[0] : null
+  )
+  const releaseTokens = useReleaseTokens()
+
+  const totalReadyToRelease = useMemo(() => {
+    return grants
+      .map((grant) => grant.readyToRelease)
+      .reduce((previous, current) => add(previous, current), 0)
+      .toString()
+  }, [grants])
+
+  const onWithdrawClick = (awaitingPromise) => {
+    releaseTokens(selectedGrant, awaitingPromise)
+  }
+
+  return (
+    <>
+      <ModalHeader>Withdraw</ModalHeader>
+      <ModalBody>
+        <h3 className={"mb-1"}>Withdraw granted tokens</h3>
+        <p className={"mb-2 text-grey-70"}>
+          The following tokens are available to be released and withdrawn from
+          your token grant.
+        </p>
+        <div>
+          <OnlyIf condition={grants.length > 1}>
+            <div className="flex row center space-between">
+              <h4 className="text-grey-70">Withdraw:</h4>
+              <span className="text-grey-70 small">
+                Total:{" "}
+                <TokenAmount
+                  amount={totalReadyToRelease}
+                  token={KEEP}
+                  wrapperClassName="withdraw-granted-tokens__total-ready-to-release-amount"
+                  amountClassName="text-grey-60"
+                  symbolClassName="text-grey-60"
+                />
+              </span>
+            </div>
+          </OnlyIf>
+          <form>
+            <Accordion
+              allowZeroExpanded={grants.length > 1}
+              className={"withdraw-granted-tokens__grants-accordion"}
+              preExpanded={grants.length === 1 ? [`grant-${grants[0].id}`] : []}
```

Comment:
Can we use `selectedGrant`?

```
grants.length === 1 ? [`grant-${selectedGrant.id}`] : []
```

Reply:
Fixed in 935eb8f56df6996b5d7be470e3275885e1fa868d